lib/date.rb

```diff
@@ -0,0 +1,43 @@
+require 'pry'
+require 'time'
+require 'date'
+
+module Hotel
+  class DateRange
+    attr_reader :start_date, :end_date
+
+    def initialize(start_date, end_date)
+      @start_date = Date.parse(start_date)
+      @end_date = Date.parse(end_date)
+
+      if (@start_date && @end_date) && (@start_date >= @end_date)
+        raise ArgumentError.new("Invalid time - start time is after end time")
+      end
```

Comment:
I like that you put this check in the constructor - that means it's impossible to ever make have an invalid date range, which makes it a little easier to reason about our program.